pandas/io/parsers.py

```diff
@@ -3607,8 +3607,8 @@ def __init__(self, f, colspecs, delimiter, comment, skiprows=None, infer_nrows=1
 
         if not isinstance(self.colspecs, (tuple, list)):
             raise TypeError(
-                "column specifications must be a list or tuple, "
-                "input was a %r" % type(colspecs).__name__
+                f"column specifications must be a list or tuple, "
+                f"input was a {repr(type(colspecs).__name__)}"
```

Comment:
you don't need any of the repr except for this one as they are all repr'ing strings.

Reply:
Show I leave it like this? 
f"input was a {type(colspecs).__name__}"

Reply:
Should I change the other strings to something like this?

f"input was a {(type(colspecs).__name__)}"

Reply:
yesp (but you can leave anything that is not a string with ``repr(...)``, e.g. this one is a list